pkg/telemetry/cassandra_agent/cassandra_agent_config.go

```diff
@@ -79,7 +162,7 @@ func (c Configurator) ReconcileTelemetryAgentConfig(dc *cassdcapi.CassandraDatac
 	recRes := reconciliation.ReconcileObject(c.Ctx, c.RemoteClient, c.RequeueDelay, *desiredCm)
 	switch {
 	case recRes.IsError():
-		fallthrough
```

Comment:
@burmanm why did you change this to fallthrough? This will give you a `Done` result AFAIK, which is wrong if the ConfigMap reconciliation has failed.